x-pack/platform/plugins/shared/triggers_actions_ui/public/application/sections/rule_details/components/rule_actions.test.tsx

```diff
@@ -74,26 +71,25 @@ describe('Rule Actions', () => {
       { id: '.index', iconClass: 'indexOpen' },
     ] as ActionTypeModel[]);
 
-    const wrapper = mount(
+    return render(
       <RuleActions ruleActions={ruleActions} actionTypeRegistry={actionTypeRegistry} />
     );
-    await act(async () => {
-      await nextTick();
-      wrapper.update();
-    });
-    return wrapper;
   }
 
   it("renders rule action connector icons for user's selected rule actions", async () => {
-    const wrapper = await setup();
+    const { container } = await setup();
     expect(mockedUseFetchRuleActionConnectorsHook).toHaveBeenCalledTimes(1);
-    expect(
-      wrapper.find('[data-euiicon-type]').length - wrapper.find('[data-euiicon-type="bell"]').length
-    ).toBe(2);
-    expect(wrapper.find('[data-euiicon-type="logsApp"]').length).toBe(1);
-    expect(wrapper.find('[data-euiicon-type="logoSlack"]').length).toBe(1);
-    expect(wrapper.find('[data-euiicon-type="index"]').length).toBe(0);
-    expect(wrapper.find('[data-euiicon-type="email"]').length).toBe(0);
+
+    // Check that the expected icons are present using data-euiicon-type attribute
+    const logsAppIcons = container.querySelectorAll('[data-euiicon-type="logsApp"]');
+    const slackIcons = container.querySelectorAll('[data-euiicon-type="logoSlack"]');
+    const indexIcons = container.querySelectorAll('[data-euiicon-type="index"]');
+    const emailIcons = container.querySelectorAll('[data-euiicon-type="email"]');
+
+    expect(logsAppIcons).toHaveLength(1);
+    expect(slackIcons).toHaveLength(1);
+    expect(indexIcons).toHaveLength(0);
+    expect(emailIcons).toHaveLength(0);
```

Comment:
I don't think the transition from Enzyme was done correctly here.

The setup function doesn't make sense in the RTL context.

This should probably be converted to a render, and then `screen` should be used for the queries.